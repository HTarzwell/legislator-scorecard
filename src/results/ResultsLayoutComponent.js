import React, { PropTypes } from 'react'

import axios from 'axios'
import LegislatorPageComponent from './LegislatorPageComponent'
import TestData from './test_data.json'

export default class ResultsLayoutComponent extends React.Component {
  constructor (props) {
    super(props)
    this.getLegislatorData = this.getLegislatorData.bind(this)
  }

  state = {
    data : undefined,
    activeTab : 'upper'
  }

  getLegislatorData (address) {
    let apiEndpoint
    if (process.env.NODE_ENV === 'production') {
      apiEndpoint = 'https://progressive-mass.herokuapp.com/local-legislators'
    } else {
      apiEndpoint = 'http://localhost:4000/local-legislators'
    }
    axios.get(apiEndpoint, {
      params : {
        address
      }
    })
  .then((response) => {
    this.setState({
      data : response.data
    })
  })
  }

  componentDidMount () {
    const address = this.props.match.params.address
    // sets state with proper values for this senator
    this.getLegislatorData(address)
  }

  render () {
    if (!this.state.data) { return (<div>loading...</div>) }

    const legislatorData = this.state.activeTab === 'upper'
    ? this.state.data.upper
    : this.state.data.lower

    return (<div className=''>
      <ul className='nav nav-pills nav-justified'>
        <li className='nav-item'>
          <a href='#'
            className={`nav-link ${this.state.activeTab === 'upper' ? 'active' : ''}`}
            onClick={(e) => { e.preventDefault(); this.setState({ activeTab : 'upper' }) }}
          >
            Your Senator
          </a>
        </li>
        <li className='nav-item'>
          <a href='#'
            className={`nav-link ${this.state.activeTab === 'lower' ? 'active' : ''}`}
            onClick={(e) => { e.preventDefault(); this.setState({ activeTab : 'lower' }) }}>

            Your House Rep
          </a>
        </li>
      </ul>
      <LegislatorPageComponent
        data={legislatorData.data}
        legislator={legislatorData.legislator}
        chamber={this.state.activeTab}
        rating={legislatorData.rating}
      />
    </div>)
  }
}

ResultsLayoutComponent.propTypes = {
  match : PropTypes.object.isRequired
}
