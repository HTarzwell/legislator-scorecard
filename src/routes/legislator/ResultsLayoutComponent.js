import React, { PropTypes } from 'react'
import Tabs from 'react-responsive-tabs'
import axios from 'axios'

import LegislatorPageComponent from './LegislatorPageComponent'
import LoadingComponent from './LoadingViewComponent'
import ErrorViewComponent from './ErrorViewComponent'

export default class ResultsLayoutComponent extends React.Component {
  constructor (props) {
    super(props)
    this.getLegislatorData = this.getLegislatorData.bind(this)
  }

  state = {
    data : undefined,
    error : false
  }

  getLegislatorData (address) {
    let apiEndpoint
    if (process.env.NODE_ENV === 'production') {
      apiEndpoint = 'https://progressive-mass.herokuapp.com/local-legislators'
    } else {
      apiEndpoint = 'http://localhost:4000/local-legislators'
    }
    axios.get(apiEndpoint, {
      params : {
        address
      }
    })
  .then((response) => {
    this.setState({
      data : response.data
    })
  }, (error) => {
    if (error.response) {
      this.setState({ error : error.response.data, data : {} })
    } else {
      this.setState({ error : error.message, data : {} })
    }
    console.log(error.message)
  })
  }

  componentDidMount () {
    const address = this.props.match.params.address
    // sets state with proper values for this senator
    this.getLegislatorData(address)
  }

  render () {
    if (this.state.error) { return <ErrorViewComponent error={this.state.error} /> }
    if (!this.state.data) { return <LoadingComponent /> }

    const tabItems = [
      {
        title : 'Your House Rep',
        component : (
          <LegislatorPageComponent
            data={this.state.data.lower}
            legislator={this.state.data.lower.legislator}
            chamber='lower'
            rating={this.state.data.lower.rating}
            />)
      },
      {
        title : 'Your Senator',
        component : (
          <LegislatorPageComponent
            data={this.state.data.upper}
            legislator={this.state.data.upper.legislator}
            chamber='upper'
            rating={this.state.data.upper.rating}
          />)
      }
    ]
    .map((t) => {
      return {
        title: t.title,
        getContent: () => t.component
      }
    })

    return (
      <div className='mt-5'>
        <Tabs items={tabItems}
          showMore={false}
          transform={false}
        />
      </div>
    )
  }
}

ResultsLayoutComponent.propTypes = {
  match : PropTypes.object.isRequired
}
