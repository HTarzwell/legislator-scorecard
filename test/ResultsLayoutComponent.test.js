import React from 'react'
import ResultsLayoutComponent from './../src/routes/legislator/ResultsLayoutComponent'
import ShallowRenderer from 'react-test-renderer/shallow'
import Renderer from 'react-test-renderer'
import { MemoryRouter } from 'react-router-dom'
import axios from 'axios'

const shallowRenderer = new ShallowRenderer()

it('renders correctly', () => {
  const tree = shallowRenderer
    .render(
        <ResultsLayoutComponent/>
    )
  expect(tree).toMatchSnapshot()
})

const fakeData = {response : {data : {}}}
axios.get = jest.fn(()=>{
  return new Promise((resolve)=>{
    resolve(fakeData)
  })
});

it('requests legislator data depending on the address', ()=>{

  const tree = Renderer
    .create(
        <ResultsLayoutComponent match={{params : { address : 'fakeAddress'}}}/>
    )
  expect(tree).toMatchSnapshot()
  expect(axios.get.mock.calls.length).toBe(1)
  expect(axios.get.mock.calls[0][0]).toBe("http://localhost:4000/local-legislators")
  expect(axios.get.mock.calls[0][1]).toEqual({"params": {"address": "fakeAddress"}})


})