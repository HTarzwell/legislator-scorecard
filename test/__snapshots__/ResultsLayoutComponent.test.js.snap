// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `<LoadingComponent />`;

exports[`requests legislator data depending on the address 1`] = `
<div
  className="loading blue-background d-flex align-items-center justify-content-center"
>
  <h1>
    loading...
  </h1>
</div>
`;
